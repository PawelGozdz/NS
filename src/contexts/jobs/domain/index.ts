export * from './job-position';
export * from './job-user-profile';
export * from './jobs';
export * from './skill';
