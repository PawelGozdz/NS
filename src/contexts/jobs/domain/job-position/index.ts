export * from './errors';
export * from './events';
export * from './job-position-command-repository.interface';
export * from './job-position-entity.fixture';
export * from './job-position-query-repository.interface';
export * from './job-position.entity';
export * from './job-position.snapshot';
