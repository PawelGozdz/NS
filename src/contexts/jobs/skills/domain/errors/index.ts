export * from './skill-already-exists.error';
export * from './skill-not-found.error';
