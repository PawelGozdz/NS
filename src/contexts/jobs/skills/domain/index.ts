export * from './errors';
export * from './events';
export * from './skill-command-repository.interface';
export * from './skill-entity.fixture';
export * from './skill-query-repository.interface';
export * from './skill.entity';
