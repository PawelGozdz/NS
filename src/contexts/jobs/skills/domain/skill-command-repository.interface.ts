import { ISkillCreateData, Skill } from './skill.entity';

export abstract class ISkillsCommandRepository {
  abstract save(skill: ISkillCreateData): Promise<{ id: number }>;

  abstract getOneById(id: number): Promise<Skill | undefined>;

  abstract getOneByNameAndCategoryId(name: string, categoryId: number): Promise<Skill | undefined>;
}
