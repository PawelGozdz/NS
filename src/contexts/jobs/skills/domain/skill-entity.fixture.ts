/* eslint-disable @typescript-eslint/strict-boolean-expressions */
import { Skill } from './skill.entity';

type Input = {
  id?: number;
  name?: string;
  description?: string;
  categoryId?: number;
};

export class SkillEntityFixtureFactory {
  public static create(overrides?: Input): Skill {
    const id = overrides?.id ?? 1;
    const name = overrides?.name ? overrides.name : 'Grocery';
    const description = overrides?.description ? overrides.description : 'Grocery Skill';
    const categoryId = overrides?.categoryId ? overrides.categoryId : 3;

    return new Skill({
      id,
      name,
      description,
      categoryId,
    });
  }
}
