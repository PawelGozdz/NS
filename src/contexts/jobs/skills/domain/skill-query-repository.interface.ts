import { ISkillsQueryParams } from '@app/core';

export type SkillInfo = {
  id: number;
  name: string;
  context: string;
  parentId: number | null;
  description: string | null;
  categoryId: number;
};

export abstract class ISkillsQueryRepository {
  abstract getManyBy(query: ISkillsQueryParams): Promise<SkillInfo[]>;
}
