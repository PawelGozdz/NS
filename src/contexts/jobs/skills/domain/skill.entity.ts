export class Skill {
  id: number;

  name: string;

  description?: string | null;

  parentId?: number | null;

  context: string;

  categoryId: number;

  constructor(props: Skill) {
    this.id = props.id;
    this.name = props.name;
    this.description = props.description ?? null;
    this.parentId = props.parentId ?? null;
    this.context = props.context;
    this.categoryId = props.categoryId;
  }

  public static create({
    id,
    name,
    description,
    parentId,
    context,
    categoryId,
  }: {
    id: number;
    name: string;
    description?: string | null;
    parentId?: number | null;
    context: string;
    categoryId: number;
  }): Skill {
    const skill = new Skill({
      id,
      name,
      description: description ?? null,
      parentId: parentId ?? null,
      context,
      categoryId,
    });

    return skill;
  }
}

export type ISkillCreateData = {
  name: string;
  description?: string | null;
  parentId?: number | null;
  context: string;
  categoryId: number;
};

export type ISkillUpdateData = {
  id: number;
  name?: string;
  description?: string | null;
  context?: string;
  parentId?: number | null;
  categoryId?: number;
};
