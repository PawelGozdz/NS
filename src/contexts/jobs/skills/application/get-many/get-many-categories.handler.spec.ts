import { createMock } from '@golevelup/ts-jest';
import { Test } from '@nestjs/testing';
import { PinoLogger } from 'nestjs-pino';

import { ISkillsQueryRepository, SkillInfo } from '../../domain';
import { GetManySkillsHandler } from './get-many-skills.handler';
import { GetManySkillsQuery } from './get-many-skills.query';

describe('GetManySkillsHandler', () => {
  let handler: GetManySkillsHandler;
  let skillQueryRepositoryMock: jest.Mocked<ISkillsQueryRepository>;

  beforeEach(async () => {
    skillQueryRepositoryMock = createMock<ISkillsQueryRepository>();

    const app = await Test.createTestingModule({
      providers: [
        GetManySkillsHandler,
        {
          provide: ISkillsQueryRepository,
          useValue: skillQueryRepositoryMock,
        },
        {
          provide: PinoLogger,
          useValue: createMock<PinoLogger>(),
        },
      ],
    }).compile();

    handler = app.get(GetManySkillsHandler);
  });

  const query: GetManySkillsQuery = {
    _filter: {
      name: 'it',
    },
  };

  const categoryId = 2;

  const skills: SkillInfo[] = [
    {
      id: 2,
      name: 'name',
      description: 'description',
      categoryId,
    },
  ];

  it('should get many skills', async () => {
    // Arrange
    skillQueryRepositoryMock.getManyBy.mockResolvedValueOnce(skills);

    // Act
    const result = await handler.execute(query);

    // Assert
    expect(result.length).toBe(1);
    expect(result).toMatchSnapshot();
  });
});
