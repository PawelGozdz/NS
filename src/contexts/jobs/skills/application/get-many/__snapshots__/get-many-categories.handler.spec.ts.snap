// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GetManySkillsHandler should get many skills 1`] = `
[
  {
    "categoryId": 2,
    "description": "description",
    "id": 2,
    "name": "name",
  },
]
`;
